### ISSUE TITLE
Improve batchnorm docs.

### ISSUE THREAD

**Issue by levskaya** [google/flax#1489]
Based on some conversations I've had with users and their frustrations, we should: 
 - add a small invocation usage example for batchnorm inside the batchnorm docstring - showing the use of `mutable=['batch_stats']`, etc.
 - either add BN to the mnist example, or make a small intro document outlining how to use BN in a training loop - given that it's the most likely "state variable" usage that people will encounter we should have a small, complete, and explicit example of using it that's quicker to read than the full imagenet example.

**Comment by levskaya** [comment]
Added a docstring note at #1490

**Comment by levskaya** [comment]
also see #932 for old notes

**Comment by marcvanzee** [comment]
@levskaya given that you improved the docstring, I guess what remains is to write some intro doc. Shall we then merge this issue with the old one at #932?

**Comment by cgarciae** [comment]
Is `mutable=True` to mutate everything no longer valid?

**Comment by avital** [comment]
@cgarciae that should work fine, but generally it's somewhat of an antipattern. Because if folks then add another collection, e.g. for intermediate outputs, users will probably want to be explicit about whether they expect those to be mutated. Otherwise you may end up with an accidental footgun where you weren't aware that things are mutating and you're ignoring the new output collections, or leaking memory keeping around those buffers if you don't use them (meaning that XLA won't know to run DCE on them, etc)

**Comment by avital** [comment]
@marcvanzee yes, I will close this and reference it from #932

**Comment by cgarciae** [comment]
Hey @avital ! I was having an error with `mutable=True` so I arrived here.

Also, when you pass `mutable=["params", "batch_stats"]` BatchNorm returns the initial batch_stats, is this expected?

**Comment by avital** [comment]
@cgarciae -- let's take this to a new thread on github discussions? (in short: with batch norm you have to pass `use_running_average=False` in either the constructor or `__call__` for it to update the stats.
